x-pack/plugins/cases/server/saved_object_types/migrations/cases.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { addOwnerToSO, SanitizedCaseOwner } from '.';
+import {
+  SavedObjectUnsanitizedDoc,
+  SavedObjectSanitizedDoc,
+  SavedObjectReference,
+} from '../../../../../../src/core/server';
+import { ACTION_SAVED_OBJECT_TYPE } from '../../../../actions/server';
+import { ESConnectorFields, pushConnectorIdReferenceName } from '../../services';
+import { ConnectorTypes, CaseType, noneConnectorId } from '../../../common';
+import { transformConnectorIdToReference } from './utils';
+
+interface UnsanitizedCaseConnector {
+  connector_id: string;
+}
+
+interface SanitizedCaseConnector {
+  connector: {
+    id: string;
+    name: string | null;
+    type: string | null;
+    fields: null | ESConnectorFields;
+  };
+}
+
+interface SanitizedCaseSettings {
+  settings: {
+    syncAlerts: boolean;
+  };
+}
+
+interface SanitizedCaseType {
+  type: string;
+}
+
+interface ConnectorIdFields {
+  connector?: { id?: string };
+  external_service?: { connector_id?: string | null } | null;
+}
+
+const transformPushConnectorIdToReference = (
```

Comment:
nit: I think is best put this function to `x-pack/plugins/cases/server/saved_object_types/migrations/utils.ts`. The same place as `transformConnectorIdToReference`.